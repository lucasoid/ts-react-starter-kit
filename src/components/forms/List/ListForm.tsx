import * as React from 'react';
import uuidv4 from 'uuid/v4';
import { Form } from '~components/forms/Form';
import { TextField } from '~components/forms/Inputs';

interface IListFormProps {
    onSubmit: Function;
    onCancel: Function;
}
export const ListForm: React.FC<IListFormProps> = props => {
    const [name, setName] = React.useState('');
    const [members, setMembers] = React.useState('');

    return (
        <Form
            onSubmit={() => {
                props.onSubmit({
                    id: uuidv4(),
                    name: name,
                    owner: 'me@domain.com',
                    members: members.split(','),
                });
            }}
            onCancel={props.onCancel}
            submitText={'Create'}
            cancelText={'Cancel'}
        >
            <TextField value={name} label="List name" onChange={setName} />
            <TextField value={'me@domain.com'} label="List owner" onChange={setName} disabled={true} />
            <TextField value={members} label="Share with" onChange={setMembers} />
        </Form>
    );
};
