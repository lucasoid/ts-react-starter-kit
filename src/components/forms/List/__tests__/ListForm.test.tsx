import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { ListForm } from '../ListForm';

jest.mock('react', () => {
    const original = jest.requireActual('react');
    return {
        __esModule: true,
        ...original,
        useState: jest.fn(),
    };
});

describe('ListForm', () => {
    it('renders a form for creating a list', () => {
        const onSubmit = jest.fn();
        const onCancel = jest.fn();
        const setter = jest.fn();
        // @ts-ignore
        React.useState.mockImplementation(() => ['value', setter]);
        let wrapper = shallow(<ListForm onSubmit={onSubmit} onCancel={onCancel} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('invokes onSubmit with some default values populated upon creation', () => {
        const onSubmit = jest.fn();
        const onCancel = jest.fn();
        const setter = jest.fn();
        // @ts-ignore
        React.useState.mockImplementation(() => ['value', setter]);
        let wrapper = mount(<ListForm onSubmit={onSubmit} onCancel={onCancel} />);
        wrapper.find('form').simulate('submit', { e: { preventDefault: () => {} } });
        expect(onSubmit.mock.calls[0][0].id).toBeTruthy();
        expect(onSubmit.mock.calls[0][0].name).toBe('value');
        expect(onSubmit.mock.calls[0][0].owner).toBeTruthy();
        expect(onSubmit.mock.calls[0][0].members).toEqual(['value']);
    });
});
