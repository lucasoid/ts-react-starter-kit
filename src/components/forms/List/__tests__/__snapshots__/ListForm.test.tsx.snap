// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListForm renders a form for creating a list 1`] = `
<Form
  cancelText="Cancel"
  onCancel={[MockFunction]}
  onSubmit={[Function]}
  submitText="Create"
>
  <TextField
    label="List name"
    onChange={[MockFunction]}
    value="value"
  />
  <TextField
    disabled={true}
    label="List owner"
    onChange={[MockFunction]}
    value="me@domain.com"
  />
  <TextField
    label="Share with"
    onChange={[MockFunction]}
    value="value"
  />
</Form>
`;
